src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseParser.cs

```diff
@@ -130,7 +109,39 @@ public static uint GetResponseHeaderNumberInfo(SafeWinHttpHandle requestHandle,
             return result;
         }
 
-        private static string GetResponseHeaderStringInfo(SafeWinHttpHandle requestHandle, uint infoLevel)
+        // Returns the first header or null if header isn't found.
+        public static string GetResponseHeaderStringInfo(SafeWinHttpHandle requestHandle, uint infoLevel)
+        {
+            uint index = 0;
+            
+            return GetResponseHeaderStringHelper(requestHandle, infoLevel, ref index);
+        }
+
+        // Returns all headers or an empty list if header isn't found.
+        public static List<string> GetResponseHeaders(SafeWinHttpHandle requestHandle, uint infoLevel)
+        {
+            uint index = 0;
+            var headerList = new List<string>();
+            string header;
+            
+            while (true)
+            {
+                header = GetResponseHeaderStringHelper(requestHandle, infoLevel, ref index);
```

Comment:
Each time `GetResponseHeaderStringHelper` is called a new `StringBuilder` is allocated (along with the `char[]` inside the builder). It'd be nice to be able to reuse the buffer across calls so there aren't so many allocations on each loop iteration, which is what #3199 was aiming to address for previous uses of `GetResponseHeaderStringInfo`.